packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -45,212 +42,156 @@ import {
-    reducer,
-    initialState
-  );
-  const { notifyStatus } = useNotifyAT();
-
-  const { get, del, post, put } = useFetchClient();
 
+  const { formatMessage } = useIntl();
+  const { formatAPIError } = useAPIErrorHandler();
+  const toggleNotification = useNotification();
   useFocusWhenNavigate();
   const { push } = useHistory();
   const { pathname } = useLocation();
+
+  const {
+    isLoading: isRBACLoading,
+    allowedActions: { canCreate, canUpdate, canDelete },
+  } = useRBAC(adminPermissions.settings.webhooks);
+  const { get, del, post, put } = useFetchClient();
+  const { notifyStatus } = useNotifyAT();
+
   const rowsCount = webhooks.length;
   const webhooksToDeleteLength = webhooksToDelete.length;
   const getWebhookIndex = (id) => webhooks.findIndex((webhook) => webhook.id === id);
 
-  useEffect(() => {
-    isMounted.current = true;
-
-    return () => {
-      isMounted.current = false;
-    };
-  }, []);
-
-  /**
-   * TODO: refactor this, but actually refactor
-   * the whole component. Needs some love.
-   */
-  useEffect(() => {
-    const fetchWebHooks = async () => {
-      try {
-        const {
-          data: { data },
-        } = await get('/admin/webhooks');
+  const fetchWebHooks = async () => {
+    try {
+      const {
+        data: { data },
+      } = await get('/admin/webhooks');
 
-        if (isMounted.current) {
-          dispatch({
-            type: 'GET_DATA_SUCCEEDED',
-            data,
-          });
-          notifyStatus('webhooks have been loaded');
-        }
-      } catch (err) {
-        console.log(err);
+      setWebhooks(data);
+      notifyStatus(
+        formatMessage({
+          id: 'Settings.webhooks.list.loading.success',
+          defaultMessage: 'Webhooks have been loaded',
+        })
+      );
 
-        if (isMounted.current) {
-          if (err.code !== 20) {
-            toggleNotification({
-              type: 'warning',
-              message: { id: 'notification.error' },
-            });
-          }
-          dispatch({
-            type: 'TOGGLE_LOADING',
-          });
-        }
-      }
-    };
-
-    if (canRead) {
-      fetchWebHooks();
-    }
-  }, [canRead, get, notifyStatus, toggleNotification]);
-
-  const handleToggleModal = () => {
-    setShowModal((prev) => !prev);
-  };
+      return data;
+    } catch (error) {
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(error),
+      });
 
-  const handleConfirmDelete = () => {
-    if (webhookToDelete) {
-      handleConfirmDeleteOne();
-    } else {
-      handleConfirmDeleteAll();
+      return null;
     }
   };
+  const QUERY_KEY = 'webhooks';
+  const { isLoading: webhooksLoading } = useQuery(QUERY_KEY, fetchWebHooks);
+  const isLoading = isRBACLoading || webhooksLoading;
 
-  const handleConfirmDeleteOne = async () => {
-    try {
-      await del(`/admin/webhooks/${webhookToDelete}`);
+  const deleteMutation = useMutation(async () => {
```

Comment:
Here we _should_ use the callbacks `onSuccess`& `onError` to do all our tidy ups, i'd probably pass the array for deleting as opposed to referencing as well, just reads simpler in one go.

This comment goes for your other mutation too, keep the `mutationFn` small and use the hooks API surface – https://tanstack.com/query/v3/docs/react/reference/useMutation